openmdao/components/interp_util/interp_akima.py

```diff
@@ -724,3 +724,566 @@ def interpolate(self, x, idx, slice_idx):
+
+        m3 = (val4 - val3) / (grid[idx + 1] - grid[idx])
+        if compute_local_train:
+            dm1_dv = np.zeros(n_this, dtype=dtype)
+            dm2_dv = np.zeros(n_this, dtype=dtype)
+            dm3_dv = np.zeros(n_this, dtype=dtype)
+            dm4_dv = np.zeros(n_this, dtype=dtype)
+            dm5_dv = np.zeros(n_this, dtype=dtype)
+            dm3_dv[idx_val4] = 1.0 / (grid[idx + 1] - grid[idx])
+            dm3_dv[idx_val3] = - dm3_dv[idx_val4]
+
+        if idx >= 2:
+            m1 = (val2 - val1) / (grid[idx - 1] - grid[idx - 2])
+            if compute_local_train:
+                dm1_dv[idx_val2] = 1.0 / (grid[idx - 1] - grid[idx - 2])
+                dm1_dv[idx_val1] = - dm1_dv[idx_val2]
+
+        if idx >= 1:
+            m2 = (val3 - val2) / (grid[idx] - grid[idx - 1])
+            if compute_local_train:
+                dm2_dv[idx_val3] = 1.0 / (grid[idx] - grid[idx - 1])
+                dm2_dv[idx_val2] = - dm2_dv[idx_val3]
+
+        if idx < ngrid - 2:
+            m4 = (val5 - val4) / (grid[idx + 2] - grid[idx + 1])
+            if compute_local_train:
+                dm4_dv[idx_val5] = 1.0 / (grid[idx + 2] - grid[idx + 1])
+                dm4_dv[idx_val4] = - dm4_dv[idx_val5]
+
+        if idx < ngrid - 3:
+            m5 = (val6 - val5) / (grid[idx + 3] - grid[idx + 2])
+            if compute_local_train:
+                dm5_dv[idx_val6] = 1.0 / (grid[idx + 3] - grid[idx + 2])
+                dm5_dv[idx_val5] = - dm5_dv[idx_val6]
+
+        if idx == 0:
+            m2 = 2 * m3 - m4
+            m1 = 2 * m2 - m3
+            if compute_local_train:
+                dm2_dv = 2.0 * dm3_dv - dm4_dv
+                dm1_dv = 2.0 * dm2_dv - dm3_dv
+
+        elif idx == 1:
+            m1 = 2 * m2 - m3
+            if compute_local_train:
+                dm1_dv = 2.0 * dm2_dv - dm3_dv
+
+        elif idx == ngrid - 3:
+            m5 = 2 * m4 - m3
+            if compute_local_train:
+                dm5_dv = 2.0 * dm4_dv - dm3_dv
+
+        elif idx == ngrid - 2:
+            m4 = 2 * m3 - m2
+            m5 = 2 * m4 - m3
+            if compute_local_train:
+                dm4_dv = 2.0 * dm3_dv - dm2_dv
+                dm5_dv = 2.0 * dm4_dv - dm3_dv
+
+        # Calculate cubic fit coefficients
+        if compute_local_train:
+            w2, dw2_dv = abs_smooth_1d(m4 - m3, dm4_dv - dm3_dv, delta_x=delta_x)
+            w31, dw31_dv = abs_smooth_1d(m2 - m1, dm2_dv - dm1_dv, delta_x=delta_x)
+        else:
+            w2 = abs_smooth_1d(m4 - m3, delta_x=delta_x)
+            w31 = abs_smooth_1d(m2 - m1, delta_x=delta_x)
+
+        b = 0.5 * (m2 + m3)
+        if compute_local_train:
+            db_dv = 0.5 * (dm2_dv + dm3_dv)
+
+        # We need to suppress some warnings that occur when we divide by zero.  We replace all
+        # values where this happens, so it never affects the result.
+        old_settings = np.seterr()
+        np.seterr(invalid='ignore', divide='ignore')
```

Comment:
This is fine but you might not know that a context manager is now available in numpy for this behavior.

https://numpy.org/doc/stable/reference/generated/numpy.errstate.html

Reply:
Oh yeah. Actually, I don't think I need to do this anymore, because we are working with a scalar here instead of a matrix.